easybuild/framework/easyconfig/easyconfig.py

```diff
@@ -491,10 +491,19 @@ def dump(self, fp):
 
         last_keys = ['sanity_check_paths', 'moduleclass']
 
+        orig_enable_templating = self.enable_templating
+        self.enable_templating = False # templated values should be dumped unresolved
+
         # build dict of default values
         default_values = dict([(key, DEFAULT_CONFIG[key][0]) for key in DEFAULT_CONFIG])
         default_values.update(dict([(key, self.extra_options[key][0]) for key in self.extra_options]))
 
+        self.generate_template_values()
+        templ_const = dict([(const[1], const[0]) for const in TEMPLATE_CONSTANTS])
+        templ_val = dict([(self.template_values[key], key) for key in self.template_values if len(self.template_values[key]) > 2])
+
+        exclude_keys = ['name', 'version', 'description', 'homepage', 'toolchain'] # values will not be templated for these keys
```

Comment:
also, this should either move to `replace_templates`, or be defined as a constant list (`EXCLUDED_KEYS_REPLACE_TEMPLATES` or somesuch) on top of this module